### PR TITLE
Update ComposeObjectTemplate and update MockTests

### PR DESCRIPTION
**Pull request by danbroudy**
This change will make implementing the blobstore multipart uploads using simple uploads and compose much cleaner.

### CODE REVIEW

**Comment by danbroudy** [comment]
ping

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 1]
```diff
@@ -14,25 +14,102 @@
...
+
+   ComposeObjectTemplate() {
+   }
+
+   public static class Builder {
+      private ImmutableList<SourceObject> sourceObjects;
+      private ObjectTemplate destination;
+
+      public Builder(){
+      }
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Should this be package protected and have a `builder` static method?

File: google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/internal/BaseGoogleComputeEngineApiLiveTest.java [hunk 2]
```diff
@@ -44,7 +44,7 @@
 public class BaseGoogleComputeEngineApiLiveTest extends BaseApiLiveTest<GoogleComputeEngineApi> {
 
    protected static final String ZONE_API_URL_SUFFIX = "/zones/";
-   protected static final String DEFAULT_ZONE_NAME = "us-central1-a";
+   protected static final String DEFAULT_ZONE_NAME = "us-central1-f";
```
Comment (andrewgaul) [inline google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/internal/BaseGoogleComputeEngineApiLiveTest.java]: Intentional?

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 3]
```diff
@@ -14,25 +14,102 @@
...
+
+      public Builder fromGoogleCloudStorageObject(List<GoogleCloudStorageObject> objects){
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Take a `Collection`?

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 4]
```diff
@@ -14,25 +14,102 @@
...
+         ArrayList<SourceObject> sourceObjects = new ArrayList<SourceObject>();
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Why not use `ImmutableList.builder`?

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 5]
```diff
@@ -14,25 +14,102 @@
...
+         for (String name : SourceObjectNames){
+            sourceObjects.add(SourceObject.nameOnly(name));
+         }
+         this.sourceObjects = ImmutableList.copyOf(sourceObjects);
+         return this;
+      }
+
+      public Builder destination(ObjectTemplate destination){
+         this.destination = destination;
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Check for null?

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 6]
```diff
@@ -14,25 +14,102 @@
...
+   public static class Builder {
+      private ImmutableList<SourceObject> sourceObjects;
+      private ObjectTemplate destination;
+
+      public Builder(){
+      }
+
+      public Builder fromGoogleCloudStorageObject(List<GoogleCloudStorageObject> objects){
+         ArrayList<SourceObject> sourceObjects = new ArrayList<SourceObject>();
+         for (GoogleCloudStorageObject obj : objects){
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Space before brace?

**Comment by andrewgaul** [comment]
Overall looks good but please address nits.

File: google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/internal/BaseGoogleComputeEngineApiLiveTest.java (hunk 2 above)
Comment (danbroudy) [inline google-compute-engine/src/test/java/org/jclouds/googlecomputeengine/internal/BaseGoogleComputeEngineApiLiveTest.java]: Yes, us-central1-a gets a lot of attention, its nice to mix it up.

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java (hunk 6 above)
Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Done.

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java (hunk 5 above)
Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Done.

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java (hunk 4 above)
Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Done.

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java (hunk 3 above)
Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Done.

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java (hunk 1 above)
Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: I don't doubt there is a better way. Im not sure what you are envisioning, are you suggesting I put a  static `builder` method be in the ComposeObjectTemplate class and not ComposeObjectTemplate.Builder class and make the current `public Builder()` method package protected?

File: google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java [hunk 7]
```diff
@@ -14,25 +14,106 @@
...
    }
+
+   ComposeObjectTemplate() {
+   }
+
+   public static class Builder {
+      private ImmutableList<SourceObject> sourceObjects;
+      private ObjectTemplate destination;
+
+      public Builder() {
```
Comment (andrewgaul) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: > I don't doubt there is a better way. Im not sure what you are envisioning, are you suggesting I put a  static `builder` method be in the ComposeObjectTemplate class and not ComposeObjectTemplate.Builder class and make the current `public Builder()` method package protected?

Correct; we should prefer a private constructor and public static `builder` method which calls it.

Comment (danbroudy) [inline google-cloud-storage/src/main/java/org/jclouds/googlecloudstorage/domain/templates/ComposeObjectTemplate.java]: Sounds good to me, updated.

**Comment by andrewgaul** [comment]
Pushed to master as 45c6c36734c8f783ddeed5e08f48f1f5e77cda0f and 1.9.x as ab9be7bf24b9f80e71f97592a1c1b86de9f72d5f.

**Pull request merged**